### PR TITLE
FIX: sparse dropout disable during eval in NGCF and GCMC

### PR DESCRIPTION
**Pull request by linzihan-backforward**

### CODE REVIEW

**Comment by deklanw** [comment]
Hello,

Doesn't this solution require that every algorithm which uses SparseDropout has to both reimplement the class and remember to turn it off in `forward`?

If we instead created a single SparseDropout class in a util file somewhere couldn't we both: 1) reuse it across algorithms, 2) turn off the dropout from within the module itself. This also matches the behavior of regular Dropout. Something like:

```python

class SparseDropout(nn.Module):
    """
    This is a Module that execute Dropout on Pytorch sparse tensor.
    """

    def __init__(self, p=0.5):
        super(SparseDropout, self).__init__()
        # p is ratio of dropout
        # convert to keep probability
        self.kprob = 1 - p

    def forward(self, x):
        # ----------------- LOOK HERE -----------------------
        if not self.training:
            return x

        mask = ((torch.rand(x._values().size()) +
                 self.kprob).floor()).type(torch.bool)
        rc = x._indices()[:, mask]
        val = x._values()[mask] * (1.0 / self.kprob)
        return torch.sparse.FloatTensor(rc, val, x.shape)
```
